compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUOps.cpp

```diff
@@ -361,9 +361,32 @@ LogicalResult ShuffleTensorOp::verifyRegions() {
 //===----------------------------------------------------------------------===//
 
 void ValueBarrierOp::build(OpBuilder &builder, OperationState &result,
-                           Value input) {
-  result.addOperands({input});
-  result.addTypes(input.getType());
+                           ValueRange input) {
+  result.addOperands(input);
+  result.addTypes(llvm::map_range(input, [](Value v) { return v.getType(); }));
+}
+
+LogicalResult ValueBarrierOp::verify() {
+  // Make sure we either have all tensors or all vectors.
+  if (hasTensorSemantics()) {
+    bool allTensor = llvm::all_of(getInputTypes(), [](ShapedType ty) {
```

Comment:
nit: you can use `llvm::isaPred` (also below for vector)
```
bool allTensor = llvm::all_of(getInputTypes(), llvm::isaPred<RankedTensorType>);
```

Reply:
done